### PR TITLE
scripts: Add script to clean up leaked osm resources

### PR DESCRIPTION
**Pull request by ksubrmnn**
Signed-off-by: Kalya Subramanian <kasubra@microsoft.com>

**Description**:
This script can be used to clean up any leaked clusterwide osm resources. These resources may be leaked if osm is not uninstalled properly such as when the osmNamespace or releaseNamespace is accidentally deleted without going through CLI or helm uninstall.

**Affected area**:

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests                  [ ]
- CI System              [ ]
- Performance            [ ]
- Other                  [X]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?
No

### CODE REVIEW

File: scripts/osm-cleanup.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+ENABLE_FLUENTBIT="${ENABLE_FLUENTBIT:-false}"
+DEPLOY_PROMETHEUS="${DEPLOY_PROMETHEUS:-true}"
+
+
+helm template "$MESH_NAME" ./charts/osm \
+    --set OpenServiceMesh.osmNamespace="$K8S_NAMESPACE" \
+    --set OpenServiceMesh.deployGrafana="$DEPLOY_GRAFANA" \
+    --set OpenServiceMesh.enableFluentbit="$ENABLE_FLUENTBIT" \
+    --set OpenServiceMesh.deployPrometheus="$DEPLOY_PROMETHEUS" \
+    --output-dir ./out \
```
Comment (shashankram) [inline scripts/osm-cleanup.sh:20]: Change to `/tmp` to avoid unnecessary untracked files within the repo? Alternative is to `rm -rf ./out` at the end.

Comment (nojnhuh) [reply scripts/osm-cleanup.sh:20]: Or `helm template ... | kubectl delete -f -` to avoid an intermediate directory altogether?

Comment (ksubrmnn) [reply scripts/osm-cleanup.sh]: Used Jon's suggestion

File: docs/troubleshooting/uninstall.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+In the `./scripts` directory we have included a helper script to delete those leaked resources: `./scripts/osm-cleanup.sh`
+
+To run the script, create a `.env` environment variable file to set the values specified at the top of the script. These values should match the values used to deploy the mesh.
+
+After running the script, you may see some errors like this.
+```
+Error from server (NotFound): error when deleting "STDIN": services "osm-controller" not found
+```
+
+Those errors should be okay as long as the environment variables were set properly. They indicate that the specified resources have been deleted prior to running the script.
```
Comment (nojnhuh) [inline docs/troubleshooting/uninstall.md:21]: You could silence these errors from the script with `kubectl delete --ignore-not-found`

**Review by nojnhuh — DISMISSED** [review]
LGTM, left one other nit.

File: docs/troubleshooting/uninstall.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# OSM Uninstall Troubleshooting Guide
+
```
Comment (shashankram) [inline docs/troubleshooting/uninstall.md:2]: Can we additionally list how to uninstall here before jumping into leaked resources directly? If there is an uninstall guide, I'd refer to it here to mention something along the lines of `"If the uninstall guide xyz was not followed, resources could be leaked..."`

File: docs/troubleshooting/uninstall.md (hunk 2 above)
Comment (ksubrmnn) [reply docs/troubleshooting/uninstall.md]: Done!

File: docs/troubleshooting/uninstall.md (hunk 3 above)
Comment (ksubrmnn) [reply docs/troubleshooting/uninstall.md:2]: Created uninstall doc and linked to it in the troubleshooting doc

**Review by nojnhuh — APPROVED** [review]

File: docs/uninstallation_guide.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Uninstallation Guide
+
+This guide describes how to uninstall Open Service Mesh (OSM) from a Kubernetes cluster using the `osm` CLI.
+
+## Prerequisites
+- Kubernetes cluster with OSM installed
+- The [osm CLI](installation_guide.md#Set-up-the-OSM-CLI)
+
+## Uninstall OSM
+Use the `osm` CLI to uninstall the OSM control plane from a Kubernetes cluster.
```
Comment (shashankram) [inline docs/uninstallation_guide.md:10]: Should we also document how to uninstall using Helm, considering the installation can be done directly via Helm?

Comment (ksubrmnn) [reply docs/uninstallation_guide.md:10]: I can follow up in a different PR so that this script can be merged.

**Review by shashankram — APPROVED** [review]

File: docs/uninstallation_guide.md (hunk 4 above)
Comment (ksubrmnn) [reply docs/uninstallation_guide.md:10]: See #2197

**Pull request merged**